pkg/controller/core/workload_controller.go

```diff
@@ -166,9 +160,51 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, nil
 	}
 
+	if ptr.Deref(wl.Spec.Active, true) {
+		var updated bool
+
+		// If a deactivated workload is re-activated we need to reset the RequeueState.
+		if workload.IsEvictedByDeactivation(&wl) && wl.Status.RequeueState != nil {
+			wl.Status.RequeueState = nil
+			updated = true
+		}
+
+		if cond := apimeta.FindStatusCondition(wl.Status.Conditions, kueue.WorkloadRequeued); cond != nil && cond.Status == metav1.ConditionFalse {
+			switch cond.Reason {
+			case kueue.WorkloadEvictedByDeactivation:
+				workload.SetRequeuedCondition(&wl, kueue.WorkloadReactivated, "The workload was reactivated", true)
+				updated = true
+			case kueue.WorkloadEvictedByPodsReadyTimeout:
+				var requeueAfter time.Duration
+				if wl.Status.RequeueState != nil && wl.Status.RequeueState.RequeueAt != nil {
+					requeueAfter = wl.Status.RequeueState.RequeueAt.Time.Sub(r.clock.Now())
+				}
+				if requeueAfter > 0 {
+					return reconcile.Result{RequeueAfter: requeueAfter}, nil
+				}
+				workload.SetRequeuedCondition(&wl, kueue.WorkloadBackoffFinished, "The workload backoff was finished", true)
+				updated = true
+			}
+		}
+
+		if updated {
+			return ctrl.Result{}, workload.ApplyAdmissionStatus(ctx, r.client, &wl, true)
+		}
+	}
+
 	cqName, cqOk := r.queues.ClusterQueueForWorkload(&wl)
 	if cqOk {
-		if updated, err := r.reconcileSyncAdmissionChecks(ctx, &wl, cqName); updated || err != nil {
+		// because we need to react to API cluster cq events, the list of checks from a cache can lead to race conditions
+		cq := kueue.ClusterQueue{}
+		if err := r.client.Get(ctx, types.NamespacedName{Name: cqName}, &cq); err != nil {
+			return ctrl.Result{}, err
+		}
+		// If stopped cluster queue is started we need to set the WorkloadRequeued condition to true.
+		if isDisabledRequeuedByClusterQueueStopped(&wl) && ptr.Deref(cq.Spec.StopPolicy, kueue.None) == kueue.None {
```

Comment:
I would suspect it does not matter what was the `StopPolicy` that lead to the setting `Requeued: false` for ClusterQueueStopped. In particular, if the `stopPolicy: HoldAndDrain`, then I believe we would set the `Requeued: false` [here](https://github.com/kubernetes-sigs/kueue/blob/f108fef22eb4532f3918754e5a08226ac366343c/pkg/controller/core/workload_controller.go#L331-L334). However, the extra check `ptr.Deref(cq.Spec.StopPolicy, kueue.None) == kueue.None` would prevent us to set `Requeued: true`.

Reply:
Oh, I think the implementation is correct (as discussed offline), we only set `Requeue: true` if `stopPolicy: None`.